#### Nigerian Inflation Data Analysis

#### Why We Want to Do This Analysis

Inflation directly affects the cost of living, purchasing power, and overall economic stability. By analyzing historical Nigerian inflation data, we can identify patterns, trends, and shifts over time. This helps us move beyond assumptions and rely on actual data to understand how prices behave across different periods.

#### Why This Analysis Is Important

Understanding inflation is critical for both individuals and businesses. For individuals, inflation determines how far income can go in meeting daily needs. For businesses, it affects pricing strategies, cost management, and long-term planning. At a broader level, inflation trends influence government policies, interest rates, and investment decisions.

Without proper analysis, decisions are often based on guesswork, which can lead to financial losses or poor planning.

#### Why This Analysis Is Useful

Analyzing this dataset allows us to:

* Track inflation trends over time (rising, falling, or stable periods)
* Compare general inflation with food inflation and core inflation
* Identify periods of economic stress or stability
* Support better financial decisions (saving, investing, budgeting)
* Build predictive models for future inflation behavior

In practical terms, this analysis can help answer questions like:

* Is inflation improving or worsening?
* Which category (food vs core) is driving inflation the most?
* How does current inflation compare to previous years?

Ultimately, this makes the data actionable, turning raw numbers into insights that can guide smarter financial decisions.


In [1]:
# import package
import pandas as pd


In [12]:
# import the dataset
df = pd.read_csv(r"C:\Users\ACER\Desktop\Portfolios\Inflation\Inflation_data.csv")
df.head(5)

,tyear,tmonth,period,allItemsYearOn,allItemsAverage,foodYearOn,foodAverage,allItemsLessFrmProdYearOn,allItemsLessFrmProdAverage,allItemsLessFrmProdAndEnergyYearOn,allItemsLessFrmProdAndEnergyAvg
0,2026,3,3/1/2026 0:00,15.38,20.05,14.31,18.21,15.66,20.86,16.21,21.09
1,2026,2,2/1/2026 0:00,15.06,21.03,12.12,19.08,15.60,21.80,15.88,22.00
2,2026,1,1/1/2026 0:00,15.10,21.97,8.89,20.29,17.18,22.58,17.72,22.84
3,2025,12,12/1/2025 0:00,15.15,23.01,10.84,22.00,18.16,23.22,18.63,23.49
4,2025,11,11/1/2025 0:00,17.33,23.13,14.21,24.36,19.84,24.08,20.59,24.38


In [14]:
# quick trend (latest vs past)
df[['period','allItemsYearOn']].head(6)

,period,allItemsYearOn
0,3/1/2026 0:00,15.38
1,2/1/2026 0:00,15.06
2,1/1/2026 0:00,15.10
3,12/1/2025 0:00,15.15
4,11/1/2025 0:00,17.33
5,10/1/2025 0:00,18.97


In [ ]:
# key inflation types
df[['allItemsYearOn','foodYearOn','allItemsLessFrmProdAndEnergyYearOn']].tail(1)

,allItemsYearOn,foodYearOn,allItemsLessFrmProdAndEnergyYearOn
278,10.6,7.5,13.6


In [16]:
# yearly average trend
df.groupby('tyear')['allItemsYearOn'].mean().tail()

tyear
2022    18.765
2023    24.520
2024    33.180
2025    23.390
2026    15.180
Name: allItemsYearOn, dtype: float64

In [18]:
# 6 month rolling average
df['forecast'] = df['allItemsYearOn'].rolling(6).mean()

df[['period','allItemsYearOn','forecast']].tail(6)

,period,allItemsYearOn,forecast
273,6/1/2003 0:00,14.0,17.100000
274,5/1/2003 0:00,8.7,15.000000
275,4/1/2003 0:00,8.3,12.450000
276,3/1/2003 0:00,5.9,10.366667
277,2/1/2003 0:00,7.3,9.516667
278,1/1/2003 0:00,10.6,9.133333


In [ ]:
# economic shock
df['change'] = df['allItemsYearOn'].diff()

# biggest spikes
df[['period','change']].sort_values('change', ascending=False).head(5)

,period,change
15,12/1/2024 0:00,7.19
262,5/1/2004 0:00,5.70
246,9/1/2005 0:00,5.70
237,6/1/2006 0:00,5.50
264,3/1/2004 0:00,5.00


In [ ]:
# salary vs inflation
salary = 175000 

latest_inflation = df['allItemsYearOn'].iloc[-1] / 100

real_value = salary / (1 + latest_inflation)

real_value

np.float64(158227.84810126582)

In [ ]:
import ollama

def get_inflation_insights(data_summary):
    # Construct a clear prompt for Qwen
    prompt = f"""
    You are an expert economic analyst. Analyze the following Nigerian inflation data:
    {data_summary}
    
    Please provide:
    1. A brief summary of the most significant inflation spikes.
    2. Potential economic drivers for these changes.
    3. Actionable advice for businesses operating in this environment.
    """
    
    response = ollama.chat(model='qwen2.5:7b', messages=[
        {'role': 'user', 'content': prompt},
    ])
    
    return response['message']['content']

# Example: Analyzing your 'Economic Shock' results
top_spikes = df[['period','change']].sort_values('change', ascending=False).head(5).to_string()
insights = get_inflation_insights(top_spikes)

print("### Qwen 2.5 Insights ###\n")
print(insights)

### Analysis of Nigerian Inflation Data

#### Summary of Significant Inflation Spikes
From the provided data, we observe that there are no significant spikes or declines; rather, the inflation rates seem to be relatively stable with minor fluctuations. The highest recorded rate is 7.19% on December 1, 2024 (period 15), and the lowest is 5.00% on March 1, 2004 (period 264). Periods 262, 246, and 237 all record a change of 5.70% and 5.50%, respectively.

#### Potential Economic Drivers for These Changes
While the provided data is limited in scope, we can infer potential drivers based on broader economic patterns and historical trends:

1. **Global Commodity Prices**: Fluctuations in global commodity prices can influence inflation, especially if Nigeria relies heavily on imported goods.
2. **Government Policy Decisions**: Fiscal policies such as changes in tax rates, government spending, or monetary policy decisions from the Central Bank of Nigeria (CBN) can impact inflation.
3. **Economic Reforms and Policies**: Implementation of economic reforms aimed at improving productivity, reducing unemployment, and increasing competition might affect prices.
4. **Monetary Supply**: The CBN's actions in regulating the money supply through interest rates or open market operations can influence inflationary pressures.
5. **Exchange Rate Fluctuations**: Changes in exchange rates, especially if Nigeria has a floating exchange rate system, can impact import costs and subsequently domestic prices.

#### Actionable Advice for Businesses Operating in This Environment

1. **Monitor Market Trends**: Stay informed about global commodity prices, local economic policies, and consumer behavior to anticipate potential changes.
2. **Diversify Supply Chains**: Reduce dependency on imported goods by diversifying supply chains or exploring local suppliers where possible.
3. **Price Management Strategies**: Implement flexible pricing strategies that can adjust quickly in response to inflationary pressures without causing customer dissatisfaction.
4. **Inventory Management**: Optimize inventory levels to minimize holding costs and reduce the risk of price increases due to inflation.
5. **Cost Control Measures**: Regularly review and control operational costs, including labor and utilities, to maintain profitability margins despite potential price hikes.
6. **Customer Engagement and Communication**: Keep customers informed about any necessary price adjustments and explain how these changes are managed within the business operations.
7. **Invest in Research and Development (R&D)**: Invest in R&D activities to innovate products or services that can offer added value, potentially commanding higher prices.
8. **Expand Market Reach**: Explore new markets or customer segments where pricing strategies might differ, thereby diversifying revenue streams.

By implementing these strategies, businesses can better navigate the economic environment and mitigate the impacts of inflation on their operations and profitability.

In [ ]:
from fpdf import FPDF
import pandas as pd

# 1. Prepare your data from inflation_analysis.ipynb
# Assuming 'df' is already loaded and processed as seen in your notebook
yearly_avg = df.groupby('tyear')['allItemsYearOn'].mean().tail()
latest_val = df['allItemsYearOn'].iloc[0]
max_spike = df['allItemsYearOn'].diff().max()

class InflationReport(FPDF):
    def header(self):
        self.set_font('Arial', 'B', 16)
        self.set_text_color(44, 62, 80)
        self.cell(0, 10, 'Nigerian Inflation Analysis Report', 0, 1, 'C')
        self.set_font('Arial', 'I', 10)
        self.cell(0, 10, 'Economic Trends & Insights (2022 - 2026)', 0, 1, 'C')
        self.ln(10)

    def footer(self):
        self.set_y(-15)
        self.set_font('Arial', 'I', 8)
        self.cell(0, 10, f'Page {self.page_no()}', 0, 0, 'C')

# 2. Create the PDF object
pdf = InflationReport()
pdf.add_page()
pdf.set_font("Arial", size=12)

# Executive Summary Section
pdf.set_font("Arial", 'B', 14)
pdf.cell(0, 10, "1. Executive Summary", 0, 1, 'L')
pdf.set_font("Arial", size=11)
summary_text = (
    "This report analyzes historical Nigerian inflation data to identify "
    "patterns and shifts over time. Understanding these trends is critical "
    "for pricing strategies, cost management, and long-term planning."
)
pdf.multi_cell(0, 7, summary_text)
pdf.ln(5)

# Key Metrics Table
pdf.set_font("Arial", 'B', 14)
pdf.cell(0, 10, "2. Key Performance Indicators", 0, 1, 'L')
pdf.set_font("Arial", 'B', 11)
pdf.cell(60, 10, "Metric", 1)
pdf.cell(60, 10, "Value", 1, 1)

pdf.set_font("Arial", size=11)
pdf.cell(60, 10, "Latest Inflation Rate", 1)
pdf.cell(60, 10, f"{latest_val}%", 1, 1)
pdf.cell(60, 10, "Highest Monthly Spike", 1)
pdf.cell(60, 10, f"{max_spike:.2f}", 1, 1)
pdf.ln(10)

# Yearly Trend Data
pdf.set_font("Arial", 'B', 14)
pdf.cell(0, 10, "3. Yearly Average Trend", 0, 1, 'L')
pdf.set_font("Arial", 'B', 11)
pdf.cell(60, 10, "Year", 1)
pdf.cell(60, 10, "Avg Inflation (YoY)", 1, 1)

pdf.set_font("Arial", size=11)
for year, value in yearly_avg.items():
    pdf.cell(60, 10, str(year), 1)
    pdf.cell(60, 10, f"{value:.2f}%", 1, 1)

# 3. Save the file
pdf.output("Inflation_Analysis_Report_v1.pdf")
print("Report generated: Inflation_Analysis_Report_v1.pdf")